File: assets/codebase/menu/DemoDiscPhysicsSceneEntry.cs
namespace city.menu {
    /// <summary>
    /// Describes one curated physics showcase scene and its Nintendo DS companion scene id.
    /// </summary>
    public sealed class DemoDiscPhysicsSceneEntry {
        /// <summary>
        /// Initializes one curated physics scene entry used by the demo-disc menu and DS scene generators.
        /// </summary>
        /// <param name="menuItemId">Stable menu item id shown in the physics scene list.</param>
        /// <param name="displayName">Human-readable scene label shown to players.</param>
        /// <param name="sceneId">Stable logical scene id used by the desktop and generic runtime flow.</param>
        /// <param name="nintendoDsSceneId">Stable logical scene id used by the Nintendo DS companion scene.</param>
        public DemoDiscPhysicsSceneEntry(string menuItemId, string displayName, string sceneId, string nintendoDsSceneId) {
            if (string.IsNullOrWhiteSpace(menuItemId)) {
                throw new ArgumentException("Menu item id must be provided.", nameof(menuItemId));
            } else if (string.IsNullOrWhiteSpace(displayName)) {
                throw new ArgumentException("Display name must be provided.", nameof(displayName));
            } else if (string.IsNullOrWhiteSpace(sceneId)) {
                throw new ArgumentException("Scene id must be provided.", nameof(sceneId));
            } else if (string.IsNullOrWhiteSpace(nintendoDsSceneId)) {
                throw new ArgumentException("Nintendo DS scene id must be provided.", nameof(nintendoDsSceneId));
            }

            MenuItemId = menuItemId;
            DisplayName = displayName;
            SceneId = sceneId;
            NintendoDsSceneId = nintendoDsSceneId;
        }

        /// <summary>
        /// Gets the stable menu item id used by the generated demo-disc menu.
        /// </summary>
        public string MenuItemId { get; }

        /// <summary>
        /// Gets the human-readable scene label shown by the menu.
        /// </summary>
        public string DisplayName { get; }

        /// <summary>
        /// Gets the logical runtime scene id used by the non-DS flow.
        /// </summary>
        public string SceneId { get; }

        /// <summary>
        /// Gets the logical runtime scene id used by the Nintendo DS companion scene.
        /// </summary>
        public string NintendoDsSceneId { get; }
    }
}

File: windows-build-20260602-cube-test-ground-high-direct/cooked/code/gameplay/DemoDiscPhysicsSceneEntry.cpp
#ifdef DrawText
#undef DrawText
#endif
#include "DemoDiscPhysicsSceneEntry.hpp"
#include "runtime/native_string.hpp"
#include "runtime/native_exceptions.hpp"
#include "DemoDiscPhysicsSceneEntry.hpp"
#include "runtime/native_exceptions.hpp"
#include "runtime/native_string.hpp"

const std::string& DemoDiscPhysicsSceneEntry::get_DisplayName()
{
return this->DisplayName;
}

const std::string& DemoDiscPhysicsSceneEntry::get_MenuItemId()
{
return this->MenuItemId;
}

const std::string& DemoDiscPhysicsSceneEntry::get_NintendoDsSceneId()
{
return this->NintendoDsSceneId;
}

const std::string& DemoDiscPhysicsSceneEntry::get_SceneId()
{
return this->SceneId;
}

DemoDiscPhysicsSceneEntry::DemoDiscPhysicsSceneEntry(std::string menuItemId, std::string displayName, std::string sceneId, std::string nintendoDsSceneId) : DisplayName(), MenuItemId(), NintendoDsSceneId(), SceneId()
{
    if (String::IsNullOrWhiteSpace(menuItemId))
    {
throw ([&]() {
auto __ctor_arg_00000000 = "Menu item id must be provided.";
auto __ctor_arg_00000001 = "menuItemId";
return new ArgumentException(__ctor_arg_00000000, __ctor_arg_00000001);
})();
    }
else {
    if (String::IsNullOrWhiteSpace(displayName))
    {
throw ([&]() {
auto __ctor_arg_00000002 = "Display name must be provided.";
auto __ctor_arg_00000003 = "displayName";
return new ArgumentException(__ctor_arg_00000002, __ctor_arg_00000003);
})();
    }
else {
    if (String::IsNullOrWhiteSpace(sceneId))
    {
throw ([&]() {
auto __ctor_arg_00000004 = "Scene id must be provided.";
auto __ctor_arg_00000005 = "sceneId";
return new ArgumentException(__ctor_arg_00000004, __ctor_arg_00000005);
})();
    }
else {
    if (String::IsNullOrWhiteSpace(nintendoDsSceneId))
    {
throw ([&]() {
auto __ctor_arg_00000006 = "Nintendo DS scene id must be provided.";
auto __ctor_arg_00000007 = "nintendoDsSceneId";
return new ArgumentException(__ctor_arg_00000006, __ctor_arg_00000007);
})();
    }
}
}
}
this->MenuItemId = menuItemId;
this->DisplayName = displayName;
this->SceneId = sceneId;
this->NintendoDsSceneId = nintendoDsSceneId;
}

